jobs/sysaddons_updates.py

```diff
@@ -0,0 +1,148 @@
+        "User-Agent": "BuildHub;storage-team@mozilla.com"
+    }
+    with async_timeout.timeout(TIMEOUT_SECONDS):
+        async with session.get(url, headers=headers, timeout=None) as response:
+            try:
+                xml_content = await response.text()
+                addons = parse_response(xml_content)
+                return addons
+            except aiohttp.ClientError as e:
+                raise ValueError("Could not fetch %s: %s" % (url, e))
+
+
+def parse_response(xml_content):
+    # <?xml version="1.0"?>
+    # <updates>
+    #   <addons>
+    #     <addon id="flyweb@mozilla.org" URL="https://ftp.mozilla.org/pub/system-addons/flyweb/flyweb@mozilla.org-1.0.xpi" hashFunction="sha512" hashValue="abcdef123" size="1234" version="1.0"/>
+    #     <addon id="pocket@mozilla.org" URL="https://ftp.mozilla.org/pub/system-addons/pocket/pocket@mozilla.org-1.0.xpi" hashFunction="sha512" hashValue="abcdef123" size="1234" version="1.0"/>
+    #   </addons>
+    # </updates>
+    root = etree.fromstring(xml_content)
+    addons = {}
+    addons_nodes = root.findall(".//addons")
+    for addon_node in addons_nodes:
+        id_ = addon_node.attrib["id"]
+        version = addon_node.attrib["version"]
+        addons[id_] = version
+    return addons
+
+
+async def main(loop):
+    parser = cli_utils.add_parser_options(
+        description="Send releases archives to Kinto",
+        default_server=DEFAULT_SERVER,
+        default_bucket=DEFAULT_BUCKET,
+        default_retry=NB_RETRY_REQUEST,
+        default_collection=DEFAULT_COLLECTION)
+
+    args = parser.parse_args(sys.argv[1:])
+
+    cli_utils.setup_logger(logger, args)
+    kinto_logger = logging.getLogger('kinto_http')
+    cli_utils.setup_logger(kinto_logger, args)
+
+    logger.info("Fetch from at {server}/buckets/{bucket}/collections/{collection}"
+                .format(**args.__dict__))
+
+    client = cli_utils.create_client_from_args(args)
+
+    filters = {
+        "source.product": "firefox"
+    }
+    records = client.get_records(**filters)
+
+    with_sysaddons = [r for r in records if r.get('systemaddons') is not None]
+    logger.info("{} entries found.".format(len(with_sysaddons)))
+
+    # Check kinto maximum batch size (can differ from AUS_BATCH_SIZE).
+    info = client.server_info()
+    kinto_batch_size = info["settings"]["batch_max_requests"]
+
+    async with aiohttp.ClientSession(loop=loop) as session:
+
+        patches = []
+        nb_batches = (len(with_sysaddons) // AUS_BATCH_SIZE) + 1
+
+        for i in range(nb_batches):
+            logger.debug("{}/{}".format(i, nb_batches))
+            records_subset = with_sysaddons[(i * AUS_BATCH_SIZE):((i + 1) * AUS_BATCH_SIZE)]
+
+            # Launch a bunch of parallel requests to AUS.
+            tasks = []
+            for record in records_subset:
+                coro = fetch_updates(session, AUS_URL, record)
+                task = asyncio.ensure_future(coro)
```

Comment:
I've meant to ask in the previous PR but forgot: is it idiomatic to use `ensure_future` instead of a simple `await` or `yield from`? I would have personally used `await`